### ISSUE TITLE
Add ability to install a nf-core module into a private module library

### ISSUE THREAD

**Issue by kenibrewer** [nf-core/tools#3174]
### Description of feature

As a maintainer of a non-nfcore modules library using nf-core tooling, I would like to be able to install nf-core modules and subworkflows into my non-nfcore modules library for use building subworkflows. 

This is currently prevented by [this check ](https://github.com/nf-core/tools/blob/930ece572bf23b68c7a7c5259e918a878ba6499e/nf_core/components/install.py#L46). It could be adjusted to check if the `org_name` is `nf_core` before failing.

**Comment by kenibrewer** [comment]
@ewels This is one of the issues.

**Comment by ewels** [comment]
Will https://github.com/nf-core/tools/pull/3083 also fix this?

**Comment by mirpedrol** [comment]
> Will https://github.com/nf-core/tools/pull/3083 also fix this?

I don't think so, we would have to adapt the command a bit to allow installing modules on a modules repo. 
But once combinations of different module sources are allowed, will you need to install an nf-core module into a non-nf-core repo?

**Comment by kenibrewer** [comment]
If you want to write an nftest case for a hybrid subworkflow then yes.

Otherwise the subworkflow can't execute in the library repo.

**Comment by mirpedrol** [comment]
oh, that's something to take into account! 👀 🚨 
So the only way in which we could test a subworkflow is to have a duplicate of the nf-core module? I am wondering if we could find an alternative, but I can't think of one right now. 
The reason is that nf-core modules will easily get out of sync every time there is an update on the nf-core repo and you don't update them on your repo. And even if we are installing the nf-core modules directly from nf-core in a pipeline (which is what #3083 will implement), this will possibly cause subworkflows to break if an updated module is used.

**Comment by awgymer** [comment]
> So the only way in which we could test a subworkflow is to have a duplicate of the nf-core module? I am wondering if we could find an alternative, but I can't think of one right now.

If #3083 is accepted you could set it up so that your tests install the subworkflow and then run the tests. This would of course necessitate having a `pipeline` type repo in your CI/CD in which to install the subworkflow into.  

> this will possibly cause subworkflows to break if an updated module is used

As it stands this sort of affects `nf-core/modules` already. If you update a module and change its input/output in a breaking way, it will break any subworkflow using it (of course this will be flagged by CI/CD).

**Comment by mirpedrol** [comment]
Hello, @nf-core/infrastructure team had several talks regarding this issue and concluded that this should not be implemented, since we want to encourage everyone to contribute to nf-core and not install an nf-core module and modify it in a private repository. 
I will close this issue, but please reopen it, or write an RFC in nf-core/proposals if you think it is still worth considering.
